src/lib/Cashlink.ts

```diff
@@ -269,16 +281,18 @@ class Cashlink {
     }
 
     public getFundingDetails(): {
-        layout: SignTransactionRequestLayout,
-        recipient: Uint8Array,
+        layout: 'cashlink',
+        recipient: Nimiq.Address,
```

Comment:
I changed it because in the new Keyguard client the SignTransactionRequest is typed more strictly with incompatible types for different layout. Therefore, the layout also needs to be specified more strictly here. While the Keyguard client is not updated in this PR, eventually this change will be necessary.
I'll remove the import for `SignTransactionLayout`.

The recipient type I changed to be easier to use in `SignTransactionLedger`.

Reply:
>While the Keyguard client is not updated in this PR
There you go. So the change has nothing to do with the Ledger support. Just wanted to make sure :wink: 

Does the new Keyguard client export the layouts as values to use here?

Reply:
The layout is not an enum in the types but just a union over the constant strings (I don't know why).
Therefore this is as good as we can do it, I guess.

Reply:
The export of these constants we should change in the KeyguardClient.